adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -0,0 +1,267 @@
+
+              ObjectMapper objectMapper = new ObjectMapper();
+              Map jsonMap = objectMapper.readValue(content, Map.class);
+
+              return jsonMap.get("Location").toString();
+            case HttpStatus.SC_TEMPORARY_REDIRECT:
+              Header locationHeader = response.getFirstHeader("Location");
+
+              return locationHeader.getValue();
+            default:
+              throw new ReplicationException(
+                  String.format("Request failed with status code: %d", status));
+          }
+        };
+
+    return sendHttpRequest(httpPut, responseHandler);
+  }
+
+  /**
+   * Executes an HTTP request
+   *
+   * @param request the HTTP request to execute
+   * @param responseHandler a value based on the response
+   * @param <T> the type expected in the response
+   * @return the response body
+   */
+  @VisibleForTesting
+  <T> T sendHttpRequest(HttpRequestBase request, ResponseHandler<T> responseHandler) {
+    try {
+      return client.execute(request, responseHandler);
+    } catch (IOException e) {
+      throw new ReplicationException(
+          String.format("Failed to send %s to remote system.", request.getMethod()), e);
+    }
+  }
+
+  /**
+   * Sends a PUT request with file data to the specified location
+   *
+   * @param createStorageRequest request containing the {@link
+   *     org.codice.ditto.replication.api.data.Resource} to create
+   * @param location the full address of the location to write the resource to
+   * @return {@code true} if successful, otherwise {@code false}
+   */
+  @VisibleForTesting
+  boolean writeFileToLocation(CreateStorageRequest createStorageRequest, String location) {
+
+    if (createStorageRequest.getResources().get(0) == null) {
+      throw new ReplicationException("Null resource encountered.");
+    }
+
+    LOGGER.debug("The location being written to is: {}", location);
+    HttpPut httpPut = new HttpPut(location);
+
+    // only a single resource is supported at this time and is the reason for always retrieving from
+    // index zero
+    Resource resource = createStorageRequest.getResources().get(0);
+
+    MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+    builder.addBinaryBody(
+        "file", resource.getInputStream(), ContentType.create(resource.getMimeType()), "file.ext");
+    HttpEntity multipart = builder.build();
+    httpPut.setEntity(multipart);
+
+    ResponseHandler<Boolean> responseHandler =
+        response -> {
+          int status = response.getStatusLine().getStatusCode();
+
+          if (status == HttpStatus.SC_CREATED) {
+            return true;
+          } else {
+            throw new ReplicationException(
+                String.format("Request failed with status code: %d", status));
+          }
+        };
+
+    return sendHttpRequest(httpPut, responseHandler);
+  }
+
+  @Override
+  public boolean updateResource(UpdateStorageRequest updateStorageRequest) {
+    return false;
```

Comment:
We certainly could have it do that now, but the implementation work for createResource is happening this week and will not be identical to calling the `createResource` method.

Reply:
At first glance, the updateResource was going to do more than call createResource. It was first going to do a GET on the filesystem and see if the file we're trying to update already exists.

But, since updateResource will only be called if the resource is indeed updated, then we will have a new resource modified timestamp and the file name will indeed be different by the suffix. IE, fileA_ts1 exists, we get update with ts2, then we create fileA_ts2 on the filesystem.

As long as we have overwrite=false when we're calling createResource from updateResource, we don't need to do any other implementation to get this to work.

Lange - I agree, though. We have a ticket for this and we should use it, because it will also be good for an integration test to make sure we aren't overwriting files. So for the combined effort I think we should keep it in the second ticket for this sprint.

Reply:
As followup, there will be a lot of overlap, but I'm going to leave that work for the separate ticket we have created, so as to avoid prolonging this pull request.